.github/workflows/static_analisys.yaml

```diff
@@ -1,5 +1,12 @@
 name: Static Analysis
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
```

Comment:
why restricting only on those two branches? could be handy while iterating in a specific branch

Reply:
mainly cause its the way on doing it with the fork-and-branch flow that is commonly used...I cant revert it back but that means:
 - We run both tests and analysis twice, one for branch, one for PR
 - We have to work on the original repo, which is risky (`git push --force` while in master.....chaos ensures)

Which is why my preferred method is to use the fork-and-branch. But as we need to use the branch from main repo for tests....it should not matter much, it will just avoid duplicating the jobs. IMHO working on the main repo is risky for me, as Im used to fuck up things in git and re-fork the whoe thing, treating my forked repo as cattle, dont care if it dies, so for me I will continue doing fork-and-branch and then send the branch against the main repo once Im happy with the result so it runs the integration tests.

If you want to iterate over a branch and run those, for the moment they can be run locally and they take about 90 seconds, which is acceptable IMO (and recommended! pushing to just check lint stuff is kind of lazy :P)

Reply:
We could also just drop the trigger on PR, and just run on branches to avoid the duplication

Reply:
> mainly cause its the way on doing it with the fork-and-branch flow that is commonly used...I cant revert it back but that means:
> 
>     * We run both tests and analysis twice, one for branch, one for PR
> 
>     * We have to work on the original repo, which is risky (`git push --force` while in master.....chaos ensures)

Don't really follow here, but I'm fine with this limitation at the end... we could also enforce protection and disable force pushes on master so they get rejected in Github

> 
> 
> Which is why my preferred method is to use the fork-and-branch. But as we need to use the branch from main repo for tests....it should not matter much, it will just avoid duplicating the jobs. IMHO working on the main repo is risky for me, as Im used to fuck up things in git and re-fork the whoe thing, treating my forked repo as cattle, dont care if it dies, so for me I will continue doing fork-and-branch and then send the branch against the main repo once Im happy with the result so it runs the integration tests.

That's ok then, it's enough for me :+1:  better have a flow that suits everyone, I'm ok to open up a PR to have static checks
> 
> If you want to iterate over a branch and run those, for the moment they can be run locally and they take about 90 seconds, which is acceptable IMO (and recommended! pushing to just check lint stuff is kind of lazy :P)

well, it assumes you have the same setup of the CI locally in every machine,  so sometime it's way much easier to check up on GH directly ;)

Reply:
doesnt really matters anymore, I moved to trigger only on branch push so it should work for master and branches. After all, we want to have integration tests always and we only get them by pushing to the branch so...we need to push to the main repo yes or yes.